audit-cli/README.md

```diff
@@ -140,6 +142,111 @@ After extraction, the code extraction report shows:
 - Code examples by language
 - Code examples by directive type
 
+#### `extract procedures`
+
+Extract unique procedures from reStructuredText files into individual files. This command parses procedures and creates
+one file per unique procedure (grouped by heading and content). Each procedure file represents a distinct piece of content,
+even if it appears in multiple selections or variations.
+
+**Use Cases:**
+
+This command helps writers:
+- Extract all unique procedures from a page for testing or migration
+- Generate individual procedure files for each distinct procedure
+- Understand how many different procedures exist in a document
+- Create standalone procedure files for reuse or testing
+- See which selections each procedure appears in
+
+**Basic Usage:**
+
+```bash
+# Extract all unique procedures from a file
+./audit-cli extract procedures path/to/file.rst -o ./output
+
+# Extract only procedures that appear in a specific selection
+./audit-cli extract procedures path/to/file.rst -o ./output --selection "driver, nodejs"
+
+# Dry run (show what would be extracted without writing files)
+./audit-cli extract procedures path/to/file.rst -o ./output --dry-run
+
+# Verbose output (shows all selections each procedure appears in)
+./audit-cli extract procedures path/to/file.rst -o ./output -v
+
+# Expand include directives inline
+./audit-cli extract procedures path/to/file.rst -o ./output --expand-includes
+```
+
+**Flags:**
+
+- `-o, --output <dir>` - Output directory for extracted procedure files (default: `./output`)
+- `--selection <value>` - Extract only procedures that appear in a specific selection (e.g., "python", "driver, nodejs")
+- `--expand-includes` - Expand include directives inline instead of preserving them
+- `--dry-run` - Show what would be extracted without writing files
+- `-v, --verbose` - Show detailed processing information including all selections each procedure appears in
+
+**Output Format:**
+
+Extracted files are named: `{heading}-{first-step-title}-{hash}.rst`
+
+The filename includes:
+- **Heading**: The section heading above the procedure
+- **First step title**: The title of the first step (for readability)
+- **Hash**: A short 6-character hash of the content (for uniqueness)
+
+Examples:
+- `before-you-begin-pull-the-mongodb-docker-image-e8eeec.rst`
```

Comment:
For legibility, could we use underscores between the different pieces of the filename? For example:
`before-you-begin_pull-the-mongodb-docker-image_e8eeec.rst`

I know that's a little odd convention-wise, but it would improve scanability if we're expecting people to reference the files.